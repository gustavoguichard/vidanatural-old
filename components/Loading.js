import { memo } from 'react'
import { FaSpinner } from 'react-icons/fa'

const Loading = ({ size = 30, absolute = true }) => {
  const positioning = absolute
    ? {
        position: 'absolute',
        top: '50%',
        left: '50%',
        marginLeft: `-${size / 2}px`,
        marginTop: `-${size / 2}px`,
      }
    : {}
  const style = {
    ...positioning,
    width: `${size}px`,
    height: `${size}px`,
  }
  return (
    <span style={style} className="spinning">
      <FaSpinner />
    </span>
  )
}

export default memo(Loading)
