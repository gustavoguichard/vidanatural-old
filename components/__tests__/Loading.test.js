import { render } from 'react-testing-library'
import Loading from 'components/Loading'

it('renders correctly', () => {
  const { container } = render(<Loading />)
  expect(container).toMatchSnapshot()
})

it('accepts props', () => {
  const { container } = render(<Loading size={60} absolute={false} />)
  expect(container).toMatchSnapshot()
})
