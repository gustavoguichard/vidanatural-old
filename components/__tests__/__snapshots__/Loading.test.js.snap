// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts props 1`] = `
<div>
  <span
    class="spinning"
    style="width: 60px; height: 60px;"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
      />
    </svg>
  </span>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <span
    class="spinning"
    style="position: absolute; top: 50%; left: 50%; margin-left: -15px; margin-top: -15px; width: 30px; height: 30px;"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
      />
    </svg>
  </span>
</div>
`;
